app/controllers/admin/products_controller.rb

```diff
@@ -75,4 +75,14 @@ def update
       product_fail("update", product)
     end
   end
+
+  def destroy
```

Comment:
Is there a way we can refactor out these destroy methods to be generic? The code is virtually the same.

Reply:
Currently, the create and update also have the same problem. I think if we did refactor this, it should be a separate branch so that we can handle all 3 of the methods at the same time.

Reply:
That sounds like the most elegant solution. Do we have time though?

Reply:
Not sure tbh. CodeClimate isn't complaining for now, but it would be nice to have.

Reply:
I think we should leave this pull request as is and potentially change this in the future if we have time. There are other more pertinent things, and as David said, CodeClimate isn't currently complaining.

Reply:
If CodeClimate isn't complaining, then I'm not going to worry about it. As you said, it's a nice to have, but not urgent